option.go

```diff
@@ -84,6 +84,10 @@ func WithAttributeFilter(filter AttributeFilter) Option {
 	return &attributeFilterOption{filterAttribute: filter}
 }
 
+func WithCustomAttributes(customAttributes CustomAttributes) Option {
+	return &customAttributesOption{customAttributes: customAttributes}
+}
+
```

Comment:
I think there should be separate options for tracing span attributes vs. metrics attributes. Metrics attributes result in more time series data, so there's a non-trivial concern with the number and cardinality of attributes for metrics that is not so much a concern with traces.

Also, exported functions need a Go doc comment.